vllm/config.py

```diff
@@ -119,35 +119,37 @@ class ModelConfig:
             override default neuron config that are specific to Neuron devices, 
             this argument will be used to configure the neuron config that 
             can not be gathered from the vllm arguments. 
+        load_params_config: Load the config from mistral format 
+            (params.json) instead of config.json.
```

Comment:
I think you made some good points on supporting checkpoint formats other than HF. Given that assumption, I think it would be best to future-proof the interface of config overrides.

I would like to propose changing this parameter to something like `config_format` that takes a value from a `ConfigFormat` enum, defaulting to ConfigFormat.HF_CONFIG or AUTO. Then `get_config()` can dispatch based on this value to the specified config parsing.

Reply:
Makes a lot of sense!